### ISSUE TITLE
Projects outside omnisharp target directory not watched

### ISSUE THREAD

**Issue by richlander** [dotnet/vscode-csharp#2986]
## Environment data
`dotnet --info` output:

```console
C:\Users\rich\testproject\consoleproj>dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100-preview5-011281
 Commit:    108e758908

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100-preview5-011281\

Host (useful for support):
  Version: 3.0.0-preview5-27613-06
  Commit:  6e994af619
```

VS Code version: 1.33.1
C# Extension version: 1.18.1-beta1

## Steps to reproduce

- Create two projects from templates
- Make P2P reference from one to another
- Change an API signature in the referenced project
- Attempt to observe new API signature in referencee project
- rebuild both projects to try to get to a clean state

## Expected  behavior

- The API usage in the referencee project shows an error (squiggles), and
- The new API usage is presented in intellisense

## Actual behavior

- Old API shape is represented
- Reboot VS Code (close / re-open).
- Correct API shape is represented

This has happened to me multiple times. I took the time now to reproduce it with the simplest repro.

Project file of referenced project:

```console
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.0</TargetFramework>
  </PropertyGroup>

</Project>
```

Project file of referencee project:

```console
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.0</TargetFramework>

    <IsPackable>false</IsPackable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.0.1" />
    <PackageReference Include="xunit" Version="2.4.0" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.0" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\consoleproj\consoleproj.csproj" />
  </ItemGroup>

</Project>
```

File layout:

```console
C:\Users\rich\testproject>dir /s /b
C:\Users\rich\testproject\consoleproj
C:\Users\rich\testproject\testproj
C:\Users\rich\testproject\consoleproj\.vscode
C:\Users\rich\testproject\consoleproj\bin
C:\Users\rich\testproject\consoleproj\consoleproj.csproj
C:\Users\rich\testproject\consoleproj\obj
C:\Users\rich\testproject\consoleproj\Program.cs
C:\Users\rich\testproject\consoleproj\.vscode\launch.json
C:\Users\rich\testproject\consoleproj\.vscode\tasks.json
C:\Users\rich\testproject\consoleproj\bin\Debug
C:\Users\rich\testproject\consoleproj\bin\Debug\netcoreapp3.0
C:\Users\rich\testproject\consoleproj\bin\Debug\netcoreapp3.0\consoleproj.deps.json
C:\Users\rich\testproject\consoleproj\bin\Debug\netcoreapp3.0\consoleproj.dll
C:\Users\rich\testproject\consoleproj\bin\Debug\netcoreapp3.0\consoleproj.exe
C:\Users\rich\testproject\consoleproj\bin\Debug\netcoreapp3.0\consoleproj.pdb
C:\Users\rich\testproject\consoleproj\bin\Debug\netcoreapp3.0\consoleproj.runtimeconfig.dev.json
C:\Users\rich\testproject\consoleproj\bin\Debug\netcoreapp3.0\consoleproj.runtimeconfig.json
C:\Users\rich\testproject\consoleproj\obj\consoleproj.csproj.nuget.cache
C:\Users\rich\testproject\consoleproj\obj\consoleproj.csproj.nuget.dgspec.json
C:\Users\rich\testproject\consoleproj\obj\consoleproj.csproj.nuget.g.props
C:\Users\rich\testproject\consoleproj\obj\consoleproj.csproj.nuget.g.targets
C:\Users\rich\testproject\consoleproj\obj\Debug
C:\Users\rich\testproject\consoleproj\obj\project.assets.json
C:\Users\rich\testproject\consoleproj\obj\Debug\netcoreapp3.0
C:\Users\rich\testproject\consoleproj\obj\Debug\netcoreapp3.0\consoleproj.AssemblyInfo.cs
C:\Users\rich\testproject\consoleproj\obj\Debug\netcoreapp3.0\consoleproj.AssemblyInfoInputs.cache
C:\Users\rich\testproject\consoleproj\obj\Debug\netcoreapp3.0\consoleproj.assets.cache
C:\Users\rich\testproject\consoleproj\obj\Debug\netcoreapp3.0\consoleproj.csproj.CoreCompileInputs.cache
C:\Users\rich\testproject\consoleproj\obj\Debug\netcoreapp3.0\consoleproj.csproj.FileListAbsolute.txt
C:\Users\rich\testproject\consoleproj\obj\Debug\netcoreapp3.0\consoleproj.csprojAssemblyReference.cache
C:\Users\rich\testproject\consoleproj\obj\Debug\netcoreapp3.0\consoleproj.dll
C:\Users\rich\testproject\consoleproj\obj\Debug\netcoreapp3.0\consoleproj.exe
C:\Users\rich\testproject\consoleproj\obj\Debug\netcoreapp3.0\consoleproj.pdb
C:\Users\rich\testproject\testproj\.vscode
C:\Users\rich\testproject\testproj\bin
C:\Users\rich\testproject\testproj\obj
C:\Users\rich\testproject\testproj\testproj.csproj
C:\Users\rich\testproject\testproj\UnitTest1.cs
C:\Users\rich\testproject\testproj\.vscode\launch.json
C:\Users\rich\testproject\testproj\.vscode\tasks.json
C:\Users\rich\testproject\testproj\bin\Debug
C:\Users\rich\testproject\testproj\bin\Debug\netcoreapp3.0
C:\Users\rich\testproject\testproj\obj\Debug
C:\Users\rich\testproject\testproj\obj\project.assets.json
C:\Users\rich\testproject\testproj\obj\testproj.csproj.nuget.cache
C:\Users\rich\testproject\testproj\obj\testproj.csproj.nuget.dgspec.json
C:\Users\rich\testproject\testproj\obj\testproj.csproj.nuget.g.props
C:\Users\rich\testproject\testproj\obj\testproj.csproj.nuget.g.targets
C:\Users\rich\testproject\testproj\obj\Debug\netcoreapp3.0
C:\Users\rich\testproject\testproj\obj\Debug\netcoreapp3.0\testproj.AssemblyInfo.cs
C:\Users\rich\testproject\testproj\obj\Debug\netcoreapp3.0\testproj.AssemblyInfoInputs.cache
C:\Users\rich\testproject\testproj\obj\Debug\netcoreapp3.0\testproj.assets.cache
C:\Users\rich\testproject\testproj\obj\Debug\netcoreapp3.0\testproj.csproj.CoreCompileInputs.cache
C:\Users\rich\testproject\testproj\obj\Debug\netcoreapp3.0\testproj.csproj.FileListAbsolute.txt
C:\Users\rich\testproject\testproj\obj\Debug\netcoreapp3.0\testproj.csprojAssemblyReference.cache
```

Visual view:

![image](https://user-images.githubusercontent.com/2608468/56093602-a1356780-5e7f-11e9-99b2-eddd344bd4fc.png)

**Comment by rchande** [comment]
@richlander Missing from your repro steps are "open the subdirectories directly containing each .csproj file in separate VS Code instances"--this seems to work as expected if I open the nearest common ancestor directory of my 2 projects.

What you're seeing here is that OmniSharp doesn't update its snapshot for referenced projects that fall outside its target directory. We could theoretically make improvements here, but we can also obviously recommend opening the common ancestor directory. Can you help me understand the scenario better?

**Comment by richlander** [comment]
If that's the recommended behavior, I'm happy to do that. I didn't know that. I will try that instead.

**Comment by richlander** [comment]
Trying that. I have two instances of Code opened:

- parent directory, enabling me to edit both app and test projects. The test project is set as the current project.
- child app directory/project.

I made an edit (added a new member in an enum) in the child app instance of Code. I went back to the parent instance and made a change in a test to take advantage of the new enum value. O# didn't see/honor it. I then opened the app file with the updated enum in the parent instance, and then O# saw it and honored the new shape of the enum from that point on.

Is it supposed to work like that?
